feat: Initial structure for the Assured Workloads example (#1)
kieras/kms-solutions: assured-workloads-example/README.md

# Assured Workloads

## Overview

> Example of Assured Workloads using Regional Control, HSM Key and Access Justification.

Reviewer: ```suggestion
Example of [Assured Workloads](https://cloud.google.com/assured-workloads/docs/overview) using [Regional Control](https://cloud.google.com/assured-workloads/docs/control-packages#regional-controls), [Cloud Hardware Security Module (HSM) Key](https://cloud.google.com/kms/docs/hsm), and [Key Access Justifications](https://cloud.google.com/assured-workloads/docs/key-access-justifications).
```

## Prerequisites

These sections describe requirements to run this example.

### Software

Install the following softwares:

> - [Google Cloud SDK](https://cloud.google.com/sdk/install)

Reviewer: is there any version constraint

Author: Works with Google Cloud SDK 484.0.0 and Terraform v1.9.1.

Author: I'll leave it "unversioned" and we can do the refinement after the example is complete, so we can test one time only.

> - [Terraform](https://www.terraform.io/downloads.html)

Reviewer: is there any version constraint?

Author: Works with Google Cloud SDK 484.0.0 and Terraform v1.9.1.

Author: I'll leave it "unversioned" and we can do the refinement after the example is complete, so we can test one time only.

### Cloud SDK configurations

> You must be authenticated in your GCP account. Run the commands below to authenticate:

Reviewer: ```suggestion
You must be authenticated in your Google Cloud Platform (GCP) account. Run the commands below to authenticate:
```

```sh
gcloud auth login
gcloud auth application-default login
```

### Service Account

> To provision the resources of this example you should use a service account.

Reviewer 2: ```suggestion
To provision the resources <"for" or "in" may fit better here> this example you should use a service account.
```

> First, make sure the following API's are enabled in the project that the service account will be created:

Reviewer: ```suggestion
First, make sure the following API's are enabled in the project in which the service account will be created:
```

> First, make sure the following API's are enabled in the project that the service account will be created:

Reviewer 2: ```suggestion
First, make sure the following APIs are enabled in the project that the service account will be created:
```

```sh
export PROJECT_ID=<YOUR-SA-PROJECT-ID>

gcloud services enable \
  iam.googleapis.com \
  cloudbilling.googleapis.com \
  serviceusage.googleapis.com \
  cloudresourcemanager.googleapis.com \
  assuredworkloads.googleapis.com \
  orgpolicy.googleapis.com \
  --project ${PROJECT_ID}
```

Use the commands below to create a [new service account](<https://cloud.google.com/iam/docs/service-accounts-create>):

```sh
export SA=<YOUR-SA-NAME>

gcloud iam service-accounts create ${SA} \
  --description="SA used to provision resources from the Assured Workloads example." \
  --display-name="Assured Workloads Provisioner" \
  --project ${PROJECT_ID}
```

> Grant the necessary roles and policy bindings to the service account:

Reviewer: ```suggestion
Grant the necessary roles to the service account:
```
